src/main/webapp/app/pages/curation/geneticTypeTabs/GeneticTypeTabs.tsx

```diff
@@ -0,0 +1,129 @@
+  const newVariantType = selected === GENETIC_TYPE.SOMATIC ? GERMLINE_PATH : SOMATIC_PATH;
+
+  useEffect(() => {
+    if (!firebaseDb) {
+      return;
+    }
+    const callbacks = [] as Unsubscribe[];
+    callbacks.push(
+      onValue(ref(firebaseDb, `${getFirebaseMetaGenePath(false, geneEntity.hugoSymbol)}/review`), snapshot => {
+        setSomaticMetaReview(snapshot.val());
+      }),
+    );
+    callbacks.push(
+      onValue(ref(firebaseDb, `${getFirebaseMetaGenePath(true, geneEntity.hugoSymbol)}/review`), snapshot => {
+        setGermlineMetaReview(snapshot.val());
+      }),
+    );
+    return () => callbacks.forEach(callback => callback?.());
+  }, []);
+
+  const geneReleaseStatus = useMemo(() => {
+    const somaticStatus = !!geneEntity?.flags?.find(flag => flag.flag === 'ONCOKB_SOMATIC' && flag.name === 'OncoKB Somatic');
+    const germlineStatus = !!geneEntity?.flags?.find(flag => flag.flag === 'ONCOKB_GERMLINE' && flag.name === 'OncoKB Germline');
+    return {
+      [GENETIC_TYPE.SOMATIC]: somaticStatus,
+      [GENETIC_TYPE.GERMLINE]: germlineStatus,
+    };
+  }, [geneEntity]);
+
+  const getGeneReleaseStatusBadge = (type: GENETIC_TYPE) => {
+    const badges: JSX.Element[] = [];
+    const sharedClassname = 'ms-2';
+    const sharedStyle: React.CSSProperties = { fontSize: '0.8rem' };
+
+    const needsReview = {
+      [GENETIC_TYPE.SOMATIC]: geneMetaReviewHasUuids(somaticMetaReview),
+      [GENETIC_TYPE.GERMLINE]: geneMetaReviewHasUuids(germlineMetaReview),
+    };
+    if (needsReview[type]) {
+      badges.push(
+        <DefaultBadge square color={'warning'} className={sharedClassname} style={sharedStyle}>
+          Needs Review
+        </DefaultBadge>,
+      );
+    }
+
+    const isGeneReleased = geneReleaseStatus[type];
+    if (isGeneReleased) {
+      // Todo: In tooltip show when gene was released
+      badges.push(
+        <DefaultBadge square color="success" className={sharedClassname} style={sharedStyle}>
+          Released
+        </DefaultBadge>,
+      );
+    } else {
+      badges.push(
+        <DefaultBadge square color={'warning'} className={sharedClassname} style={sharedStyle}>
+          Pending Release
+        </DefaultBadge>,
+      );
+    }
+
+    return <>{badges.map(badge => badge)}</>;
+  };
+
+  function handleToggle() {
+    localStorage.setItem(SOMATIC_GERMLINE_SETTING_KEY, newVariantType);
+    window.location.href = pathname.replace(currentVariantType, newVariantType);
+  }
+
+  return (
+    <div className={styles.tabs}>
+      {[GENETIC_TYPE.SOMATIC, GENETIC_TYPE.GERMLINE].map((geneOrigin, idx) => (
+        <div
+          key={idx}
+          style={{ width: '50%' }}
+          className={
+            selected === geneOrigin ? classnames(styles.tab, styles.selectedTab, 'fw-bold') : classnames(styles.tab, styles.unselectedTab)
+          }
+          onClick={handleToggle}
+        >
+          {geneOrigin.substring(0, 1).toUpperCase()}
```

Comment:
Think it may be simpler here to just hard code the values, especially since it's only two items (and as far as I know it will only ever be two). However, this preference, so feel free to ignore if you disagree.

Reply:
Yeah - I will keep it as is because I copied most of this file over from the public repository.